moda-contracts/src/CatalogFactory.sol

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.21;
+
+import "./interfaces/ICatalogFactory.sol";
+import "./interfaces/ICatalog.sol";
+import "@openzeppelin/contracts/proxy/beacon/BeaconProxy.sol";
+
+contract CatalogFactory is ICatalogFactory {
+    /// Storage Variables
+    address public immutable catalogBeacon;
+    address[] public catalogs;
+
+    /// Errors
+    error AddressCannotBeZero();
+
+    /// Constructor
+    constructor(address catalogBeacon_) {
+        if (catalogBeacon_ == address(0)) {
+            revert AddressCannotBeZero();
+        }
+        catalogBeacon = catalogBeacon_;
+    }
+
+    /// External Functions
+    /**
+     * @inheritdoc ICatalogFactory
+     */
+    function deployCatalog(
+        string calldata catalogName,
```

Comment:
I wonder if we should use the `Profile` contract instead of strings. Just something to think about

Reply:
Hmm yeah so maybe in a catalogs initialize we could call mint on a new profile token and save the token id to storage in the Catalog as reference

Reply:
I just added a feature where an owner or user with `DEFAULT_ADMIN_ROLE` if using `AccessControl` can mint / update a profile on behalf of a contract. This is important for backwards compatibility.